include/envoy/ssl/connection.h

```diff
@@ -73,6 +74,18 @@ class Connection {
    *         Returns {} if there is no local certificate, or no SAN field, or no DNS.
    **/
   virtual std::vector<std::string> dnsSansLocalCertificate() const PURE;
+
+  /**
+   * @return SystemTime the time that the peer certificate was issued and should be considered valid
+   *         from. Returns zero value SystemTime if there is no peer certificate.
+   **/
+  virtual SystemTime validFromPeerCertificate() const PURE;
```

Comment:
Thoughts on whether we want to return a system time here and below or something that is guaranteed UTC? Thoughts @PiotrSikora @lizan?